MekHQ/src/mekhq/campaign/universe/companyGeneration/CompanyGenerationPersonTracker.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2021-2022 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MekHQ.
+ *
+ * MekHQ is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MekHQ is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MekHQ. If not, see <http://www.gnu.org/licenses/>.
+ */
+package mekhq.campaign.universe.companyGeneration;
+
+import megamek.common.Entity;
+import megamek.common.annotations.Nullable;
+import mekhq.campaign.personnel.Person;
+import mekhq.campaign.universe.enums.CompanyGenerationPersonType;
+
+/**
+ * @author Justin "Windchild" Bowen
+ */
+public class CompanyGenerationPersonTracker {
```

Comment:
OK further down the code I saw that this seems a temporary person container during generation. There's probably good reason not to use the Person class itself here, maybe write that down.

Reply:
Added a comment, and it's not a person because it needs additional fields, and you may end up rerolling the person from the frontend while needing them to stay at the same spot within the list.